app/src/main/java/de/dhbw/corona_world_app/ui/map/MapFragment.java

```diff
@@ -1,28 +1,99 @@
 package de.dhbw.corona_world_app.ui.map;
 
+import android.annotation.SuppressLint;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.TextView;
+import android.webkit.ConsoleMessage;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
 
 import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
-import androidx.lifecycle.Observer;
+import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModelProvider;
 
+import java.util.concurrent.ExecutorService;
+
 import de.dhbw.corona_world_app.R;
+import de.dhbw.corona_world_app.ThreadPoolHandler;
+import de.dhbw.corona_world_app.ui.tools.LoadingScreenInterface;
 
 public class MapFragment extends Fragment {
 
+    private static final String TAG = MapFragment.class.getSimpleName();
+
     private MapViewModel mapViewModel;
 
-    public View onCreateView(@NonNull LayoutInflater inflater,
-                             ViewGroup container, Bundle savedInstanceState) {
-        mapViewModel =
-                new ViewModelProvider(this).get(MapViewModel.class);
+    MutableLiveData<String> webViewString = new MutableLiveData<>();
+
+    LoadingScreenInterface loadingScreen = new LoadingScreenInterface() {
+        @Override
+        public void startLoadingScreen() {
+
+        }
+
+        @Override
+        public void endLoadingScreen() {
+
+        }
+
+        @Override
+        public void setProgressBar(int progress, @NonNull String message) {
+
+        }
+
+        @Override
+        public int getProgress() {
+            return 0;
+        }
+    };
+
+    //todo map ISOCodes to screen names for better understanding
+    //todo WebView is not final -> more zoom, clickable tooltips with routing to statistics
+    //todo establish order
+    //loading screen will be implemented by ui-team
+    @SuppressLint("SetJavaScriptEnabled")
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        Log.v(TAG,"Starting loading screen");
+        loadingScreen.startLoadingScreen();
+        Log.v(TAG,"Creating MapFragment view");
+        mapViewModel = new ViewModelProvider(this).get(MapViewModel.class);
```

Comment:
bind viewmodel to activity or use cache directory

Reply:
i would suggest you use the cache directory

Reply:
using cache directory, cache now is no outsourced class any more -> integrated into MapViewModel and later StatisticViewModel